### DISCUSSION TITLE
time trigger with user defined function

### DISCUSSION THREAD

**Discussion by stefanuytterhoeven** [custom-components/pyscript#365] [Q&A] [answered]
question about the time trigger parameter.
it has to be a datetime, "sunset", ... 
but, is there a way to use the result of a function (datetime) .
I now have two input helper fields: input_select.starttime en input_datetime.starttime
the input_select.starttime can have values like sunset, sunrise,... so valid values for a timetrigger
but, it has also a value "usedatetime", what means, the trigger has to use the input_datetime.starttime

what i tried was to make de def get_timevalue() that return the value of the input_datetime_starttime, when the input_select.starttime == "usedatetime", else, it return the value of the input_select.starttime helperfield.

But, it seems not to work. Is is not possible? Or am i doing something wrong?

**Comment by craigbarratt** [comment, marked as answer]
The string arguments to `@time_trigger` are only evaluated once when the function is defined.  So the only way to have a dynamic trigger time is to define a `@time_trigger` function via a closure (inner function), and convert any `datetime` value to a correctly formatted string giving the desired trigger time.

**Reply by stefanuytterhoeven** [reply]
found another soluton. inserted a new inner function , with a state trigger and time trigger (startup)
When the input fields are changed, the original inner function (time trigger) is recreated.
and,.... it's working. So, in the end, every time I change an inputfield, the most inner time trigger is recreated.
Thx!
